src/Framework/AspNetCoreAnalyzers/src/SourceGenerators/PublicTopLevelProgramGenerator.cs

```diff
@@ -0,0 +1,44 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.AspNetCore.SourceGenerators;
+
+[Generator]
+public class PublicProgramSourceGenerator : IIncrementalGenerator
+{
+    private const string PublicPartialProgramClassSource = """
+// <auto-generated />
+public partial class Program { }
+""";
+
+    public void Initialize(IncrementalGeneratorInitializationContext context)
+    {
+        var internalGeneratedProgramClass = context.CompilationProvider
+            // Get the entry point associated with the compilation, this maps to the Main method definition
+            .Select((compilation, cancellationToken) => compilation.GetEntryPoint(cancellationToken))
```

Comment:
I don't know whether it matters, but it seems like all these lambdas could be static.

Reply:
Yep -- good call.